### PR TITLE
DAOS-4617 test: Fix bandit error B101 B102 B110 on python scripts

### PR DESCRIPTION
**Pull request by dinghwah**
new file: ./src/tests/ftest/security/bandit.config
modified: ./utils/sl/env_modules.py
modified: ./src/client/pydaos/daosdbm.py

Test-tag-hw-large: pr,hw,large
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

### CODE REVIEW

File: src/client/pydaos/daosdbm.py [hunk 1]
```diff
@@ -135,9 +136,9 @@ def main():
 
     print('Kvs are {}'.format(','.join(sorted(my_kv.get_kv_list()))))
 
-    if True:
-        new_migrate(my_kv)
-        return
+    #if True:
```
Comment (saurabhtandan) [inline src/client/pydaos/daosdbm.py:139]: Please remove if not needed

**Comment by johannlombardi** [comment]
@phender @saurabhtandan Please prioritize this review.

File: utils/sl/env_modules.py [hunk 2]
```diff
@@ -110,8 +110,8 @@ def _init_module_func(self, default_func):
     def _init_mpi_module(self):
         """init mpi module function"""
         try:
-            subprocess.check_call(['sh', '-l', '-c', 'module -V'],
-                                  stdout=DEVNULL, stderr=DEVNULL)
+            subprocess.check_call(['sh', '-l', '-c', 'module -V'], \
+                                  stdout=DEVNULL, stderr=DEVNULL)  #nosec
```
Comment (saurabhtandan) [inline utils/sl/env_modules.py:114]: Specific reason we have "#nosec" in the lines above?

Comment (dinghwah) [reply utils/sl/env_modules.py:114]: From testing, full path of /usr/bin/sh causing failure in build:
Test stage Build on Ubuntu 18.04 with Clang completed with status FAILURE. https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-2602/7/execution/node/198/log

File: src/client/pydaos/daosdbm.py [hunk 3]
```diff
@@ -135,9 +136,8 @@ def main():
 
     print('Kvs are {}'.format(','.join(sorted(my_kv.get_kv_list()))))
 
-    if True:
-        new_migrate(my_kv)
-        return
+    new_migrate(my_kv)
+    return
```
Comment (kjacque) [inline src/client/pydaos/daosdbm.py:140]: This is very strange - returning in the middle of the function? The original issue was not really your bug but this is an error. Should this block be moved after the other checks? Or the stuff afterward should be removed?

Comment (dinghwah) [reply src/client/pydaos/daosdbm.py:140]: Thanks for the comment, bandit caught the issue of line 138 if True: as redundant statement, we might need to revisit the script for the rest of script planed to do.

Comment (kjacque) [reply src/client/pydaos/daosdbm.py:140]: It seems like static analysis would tag this as unreachable code. Maybe @ashleypittman can comment on the intent. In this case I'd like the correct fix -- either changing the check to something meaningful, or removing the code afterward if it's not intended to be used.

Comment (ashleypittman) [reply src/client/pydaos/daosdbm.py:140]: Most of this file is scheduled for removal in 1.2, it's not currently installed as part of the RPM.

Everything after L74 could be deleted.

File: utils/sl/env_modules.py [hunk 4]
```diff
@@ -110,8 +110,8 @@ def _init_module_func(self, default_func):
     def _init_mpi_module(self):
         """init mpi module function"""
         try:
-            subprocess.check_call(['sh', '-l', '-c', 'module -V'], \
-                                  stdout=DEVNULL, stderr=DEVNULL)  #nosec
```
Comment (kjacque) [inline utils/sl/env_modules.py]: Don't we still need to specify #nosec in these cases? Or is Bandit satisfied?

Comment (dinghwah) [reply utils/sl/env_modules.py]: please see the new commit.

**Comment by dinghwah** [comment]
Functional test failures triaged from PR DAOS-4716 & DAOS_4716_2:
not related to the fix:
(1)DAOS-3945: DaosServerConfigTest.08 & Test14: Failed to stop servers:
(2)DAOS-4490: DaosCoreTest.01 No available hugepages reported in starting a daos server.
(3)DAOS-4700: DAOSCoreTest.07 Failed to start servers before format:

**Pull request merged**